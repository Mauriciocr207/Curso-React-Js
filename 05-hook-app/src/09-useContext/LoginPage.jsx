import { useContext } from "react"
import UserContext from "./context/UserContext";

export default function LoginPage() {
  const { user, setUser } = useContext(UserContext);

  return (
    <>
        <h1>Login Page</h1>
        <hr />

        <pre data-testid="LoginPage.Pre">
          { JSON.stringify(user, null, 3) }
        </pre>

        <button 
          data-testid="LoginPage.Button"
          className="bg-blue-500 p-2 rounded-lg text-white mt-5"
          onClick={() => setUser({id: 123, name: 'Mau', email: 'mau@gmail.com'})}
        >
          Iniciar Sesi√≥n
        </button>
    </>
  )
}