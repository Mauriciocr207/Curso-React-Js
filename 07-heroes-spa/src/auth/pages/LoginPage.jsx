import {
  Card,
  Input,
  Button,
  CardBody,
  CardHeader,
  Typography,
} from "@material-tailwind/react";
import { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import AuthContext from "../context/AuthContext";

export default function LoginPage() {
  const navigate = useNavigate();
  const { login } = useContext(AuthContext);
  const [ name, setName ] =  useState('');

  const handleLogin = () => {
    const lastPath = localStorage.getItem('lastPath') || "/";
    login(name);
    navigate(lastPath, {replace: true});
  };

  

  return (
    <section className="px-8">
      <div className="container mx-auto h-screen grid place-items-center">
        <Card
          className="px-10 py-16 md:px-24 md:py-14 border border-gray-300"
        >
          <CardHeader shadow={false} floated={false} className="text-center">
            <Typography
              variant="h1"
              color="blue-gray"
              className="mb-4 !text-3xl lg:text-4xl"
            >
              Inicia Sesi√≥n
            </Typography>
            <Typography className="!text-gray-600 text-[18px] font-normal md:max-w-sm">
              Comienza a administrar a tus h√©roes üòâ
            </Typography>
          </CardHeader>
          <CardBody>
            <form
              action="#"
              className="flex flex-col gap-4 md:mt-5"
              onSubmit={(e) => {
                e.preventDefault();
                handleLogin();
              }}
            >
              <div>
                <label htmlFor="email">
                  <Typography
                    variant="small"
                    color="blue-gray"
                    className="block font-medium mb-2"
                  >
                    Tu Nombre
                  </Typography>
                </label>
                <Input
                  id="name"
                  color="gray"
                  size="lg"
                  type="name"
                  name="name"
                  placeholder="Mauricio Carrillo"
                  className="w-full placeholder:opacity-100 focus:!border-t-gray-900"
                  value={name}
                  onChange={({target}) => setName(target.value)}
                />
              </div>
              <Button onClick={handleLogin} size="lg" color="gray" fullWidth>
                Iniciar Sesi√≥n
              </Button>
              <Typography
                variant="small"
                className="text-center mx-auto max-w-[19rem] !font-medium !text-gray-600"
              >
                Al iniciar sesi√≥n, est√°s de acuerdo con nuestros{" "}
                <a href="#" className="text-gray-900">
                  T√©rminos del Servicio
                </a>{" "}
                y{" "}
                <a href="#" className="text-gray-900">
                  Pol√≠ticas de Privacidad
                </a>
              </Typography>
            </form>
          </CardBody>
        </Card>
      </div>
    </section>
  );
}