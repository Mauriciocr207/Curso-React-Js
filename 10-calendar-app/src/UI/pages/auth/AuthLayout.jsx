import "notyf/notyf.min.css";

export default function AuthLayout({ children }) {
  return (
    <section className="bg-gray-100 dark:bg-gray-900">
      <div className="flex flex-col items-center justify-center px-6 py-8 mx-auto h-screen lg:py-0-w-[30rem]">
        { children }
      </div>
    </section>
  )
}