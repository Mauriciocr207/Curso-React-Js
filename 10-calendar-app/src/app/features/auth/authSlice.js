import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  isLoading: true,
  isAuthenticated: false,
  user: null,
  errorMessage: null,
}

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    login: (state, action) => {
      state.user = action.payload;
      state.isLoading = false;
      state.isAuthenticated = true;
      state.errorMessage = null;
    },
    logout: (state) => {
      state.user = null;
      state.isAuthenticated = false;
      state.isLoading = false;
    },
    load: (state) => {
      state.isLoading = true
    },
    setErrorMessage: (state, action) => {
      state.errorMessage = action.payload;
    }
  }
});

export const {
  login,
  logout,
  load,
  setErrorMessage
} = authSlice.actions

export default authSlice.reducer