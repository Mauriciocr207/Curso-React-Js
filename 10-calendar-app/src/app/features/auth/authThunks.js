import calendarApi from "../../api/calendarApi";
import { removeAllEvents } from "../calendar/calendarSlice";
import { load, login, logout, setErrorMessage } from "./authSlice"

export function startLogin({ email, pass }) {
    return async (dispatch) => {
        try {
            dispatch(load())
            
            const response = await calendarApi.post('/auth/login', { email, pass });

            const user = response.data;

            localStorage.setItem('token', user.token);
            
            dispatch(login({ name: user.name, id: user.id }));
        } catch (error) {
            if(error.name === "AxiosError") {
                dispatch(setErrorMessage("Credenciales invÃ¡lidas"));
                setTimeout(() => {
                    dispatch(setErrorMessage(null));
                }, 2000);
            }

            dispatch(logout());
        }
    }
}

export function startRegister({ name, email, pass }) {
    return async (dispatch) => {
        try {
            dispatch(load());

            const { data: user } = await calendarApi.post('/auth/register', { name, email, pass });

            localStorage.setItem('token', user.token);
            
            dispatch(login({ name: user.name, id: user.id }));
        } catch (error) {
            if(error.name === "AxiosError") {
                const { error: responseErr } = error.response.data;
                if(responseErr) {
                    dispatch(setErrorMessage(responseErr.message));
                }

                const { errors } = error.response.data;
                
                if(errors) {
                    if(errors.email) {
                        dispatch(setErrorMessage(errors.email.msg));
                    } else if(errors.name) {
                        return dispatch(setErrorMessage(errors.name.msg));
                    } else if(errors.pass) {
                        dispatch(setErrorMessage(errors.pass.msg));
                    }
                }
                
                setTimeout(() => {
                    dispatch(setErrorMessage(null));
                }, 2000);
            }

            dispatch(logout());
        }
    }    
}

export function checkAuthToken() {
    return async(dispatch) => {
        try {
            dispatch(load());
            const token = localStorage.getItem('token');
            if(!token) {
                return dispatch(logout());
            }

            const { data: user  } = await calendarApi.get('/auth/re-validate-token');

            localStorage.setItem('token', user.token);

            dispatch(login({ name: user.name, id: user.id }));
        } catch (error) {
            console.log(error);
        }
    }
}

export function startLogout() {
    return (dispatch) => {
        localStorage.removeItem('token');
        dispatch(logout());
        dispatch(removeAllEvents());
    }
}