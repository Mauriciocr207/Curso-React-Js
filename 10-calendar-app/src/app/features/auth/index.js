import authSlice from "./authSlice"

export * from "./authThunks";
export {
    authSlice
}