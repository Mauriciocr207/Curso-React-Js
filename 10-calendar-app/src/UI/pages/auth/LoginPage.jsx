import { useContext, useState } from "react";
import AuthLayout from "./AuthLayout";
import { Input } from "@nextui-org/input";
import { FaEye, FaEyeSlash } from "react-icons/fa";
import { Button } from "@nextui-org/button";
import { Spinner } from "@nextui-org/spinner";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks";
import { loginValidation } from "../../validation";
import { useDispatch, useSelector } from "react-redux";
import { startLogin } from "../../../app/features/auth";
import NotyfContext from "../../context/NotyfContext";
import { Card } from "@nextui-org/react";

const loginFields = {
  email: "",
  pass: "",
};

export default function LoginPage() {
  const { isLoading, errorMessage } = useSelector((state) => state.auth);
  const [isVisible, setIsVisible] = useState(false);
  const dispatch = useDispatch();
  const notyf = useContext(NotyfContext);
  const {
    email,
    pass,
    onInputChange,
    errors,
    isFormValid,
    formState,
    onResetForm,
  } = useForm(loginFields, loginValidation);

  const toggleVisibility = () => setIsVisible(!isVisible);

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!isFormValid) {
      console.log("invalid");
      notyf.error("Completa los campos correctamente para continuar");
      return;
    }

    dispatch(startLogin(formState));
    onResetForm();
  };

  return (
    <AuthLayout>
      <h1 className="text-3xl font-bold leading-tight tracking-tight text-gray-700 md:text dark:text-white mb-5">
        Calendar App
      </h1>
      <div className="w-full bg-white rounded-xl shadow-lg dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700">
        <div className="p-6 space-y-4 md:space-y-6 sm:p-8">
          <h2 className="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white">
            Sign in to your account
          </h2>

          {errorMessage && (
            <Card data-testid="errorMessage" className="bg-danger text-white font-semibold px-5 py-2 text-center">
              {errorMessage}
            </Card>
          )}

          <form onSubmit={handleSubmit} className="space-y-4 md:space-y-6">
            <Input
              data-testid="email"
              variant="bordered"
              type="email"
              name="email"
              label="Your Email"
              placeholder="Enter your email"
              className="w-full shadow rounded-medium"
              value={email}
              onChange={onInputChange}
              isInvalid={!!errors.email}
              errorMessage={errors.email}
            />
            <Input
              data-testid="password"
              variant="bordered"
              label="Password"
              placeholder="Enter your password"
              name="pass"
              endContent={
                <button
                  className="focus:outline-none"
                  type="button"
                  onClick={toggleVisibility}
                  aria-label="toggle password visibility"
                >
                  {isVisible ? (
                    <FaEyeSlash className="text-2xl text-default-400 pointer-events-none" />
                  ) : (
                    <FaEye className="text-2xl text-default-400 pointer-events-none" />
                  )}
                </button>
              }
              type={isVisible ? "text" : "password"}
              className="w-full shadow rounded-medium"
              value={pass}
              onChange={onInputChange}
              isInvalid={!!errors.pass}
              errorMessage={errors.pass}
            />
            <Button
              data-testid="login-button"
              type="submit"
              className="w-full text-white bg-primary-600 hover:bg-primary-700 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
            >
              {isLoading ? <Spinner size="sm" color="white" /> : "Sign in"}
            </Button>
            <p className="text-sm font-light text-gray-500 dark:text-gray-400">
              Donâ€™t have an account yet?{" "}
              <Link
                to="/auth/register"
                className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              >
                Sign up
              </Link>
            </p>
          </form>
        </div>
      </div>
    </AuthLayout>
  );
}
