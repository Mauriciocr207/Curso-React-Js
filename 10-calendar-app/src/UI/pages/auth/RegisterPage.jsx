import { useContext, useState } from "react";
import AuthLayout from "./AuthLayout";
import { FaEye, FaEyeSlash } from "react-icons/fa";
import { Input } from "@nextui-org/input";
import { Button } from "@nextui-org/button";
import { Spinner } from "@nextui-org/spinner";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks";
import { registerValidation } from "../../validation";
import { useDispatch, useSelector } from "react-redux";
import NotyfContext from "../../context/NotyfContext";
import { startRegister } from "../../../app/features/auth";
import { Card } from "@nextui-org/react";

const loginFields = {
  name: "",
  email: "",
  pass: "",
  confirmPass: "",
};

export default function RegisterPage() {
  const [passIsVisible, setPassIsVisible] = useState(false);
  const [confirmIsVisible, setConfirmIsVisible] = useState(false);
  const {
    name,
    email,
    pass,
    confirmPass,
    onInputChange,
    errors,
    isFormValid,
    formState,
    onResetForm,
  } = useForm(loginFields, registerValidation);
  const { isLoading, errorMessage } = useSelector((state) => state.auth);
  const notyf = useContext(NotyfContext);
  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!isFormValid) {
      notyf.error("Completa los campos correctamente para continuar");
      return;
    }

    dispatch(startRegister(formState));
    onResetForm();
  };

  return (
    <AuthLayout>
      <h1 className="text-3xl font-bold leading-tight tracking-tight text-gray-700 md:text dark:text-white mb-5">
        Calendar App
      </h1>
      <div className="w-full bg-white rounded-xl shadow-lg dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700">
        <div className="p-6 space-y-4 md:space-y-6 sm:p-8">
          <h2 className="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white">
            Sign in to your account
          </h2>
          {errorMessage && (
            <Card data-testid="errorMessage" className="bg-danger text-white font-semibold px-5 py-2 text-center">
              {errorMessage}
            </Card>
          )}
          <form onSubmit={handleSubmit} className="space-y-4 md:space-y-6">
            <Input
              data-testid="name"
              variant="bordered"
              type="text"
              label="Your Name"
              placeholder="Enter your name"
              className="w-full shadow rounded-medium"
              name="name"
              value={name}
              onChange={onInputChange}
              isInvalid={!!errors.name}
              errorMessage={errors.name}
            />
            <Input
              data-testid="email"
              variant="bordered"
              type="email"
              label="Your Email"
              placeholder="Enter your email"
              className="w-full shadow rounded-medium"
              name="email"
              value={email}
              onChange={onInputChange}
              isInvalid={!!errors.email}
              errorMessage={errors.email}
            />
            <Input
              data-testid="password"
              variant="bordered"
              label="Password"
              placeholder="Enter your password"
              endContent={
                <button
                  className="focus:outline-none"
                  type="button"
                  onClick={() => setPassIsVisible(!passIsVisible)}
                  aria-label="toggle password visibility"
                >
                  {passIsVisible ? (
                    <FaEyeSlash className="text-2xl text-default-400 pointer-events-none" />
                  ) : (
                    <FaEye className="text-2xl text-default-400 pointer-events-none" />
                  )}
                </button>
              }
              type={passIsVisible ? "text" : "password"}
              className="w-full shadow rounded-medium"
              name="pass"
              value={pass}
              onChange={onInputChange}
              isInvalid={!!errors.pass}
              errorMessage={errors.pass}
            />
            <Input
              data-testid="confirmPassword"
              variant="bordered"
              label="Confirm Password"
              placeholder="Confirm your password"
              endContent={
                <button
                  className="focus:outline-none"
                  type="button"
                  onClick={() => setConfirmIsVisible(!confirmIsVisible)}
                  aria-label="toggle password visibility"
                >
                  {confirmIsVisible ? (
                    <FaEyeSlash className="text-2xl text-default-400 pointer-events-none" />
                  ) : (
                    <FaEye className="text-2xl text-default-400 pointer-events-none" />
                  )}
                </button>
              }
              type={confirmIsVisible ? "text" : "password"}
              className="w-full shadow rounded-medium"
              name="confirmPass"
              value={confirmPass}
              onChange={onInputChange}
              isInvalid={!!errors.confirmPass}
              errorMessage={errors.confirmPass}
            />
            <Button
              data-testid="register-button"
              type="submit"
              className="w-full text-white bg-primary-600 hover:bg-primary-700 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
            >
              {isLoading ? (
                <Spinner size="sm" color="white" />
              ) : (
                "Create account"
              )}
            </Button>
            <p className="text-sm font-light text-gray-500 dark:text-gray-400">
              Have an account?{" "}
              <Link
                to="/auth/login"
                className="font-medium text-primary-600 hover:underline dark:text-primary-500"
              >
                Sign in
              </Link>
            </p>
          </form>
        </div>
      </div>
    </AuthLayout>
  );
}
