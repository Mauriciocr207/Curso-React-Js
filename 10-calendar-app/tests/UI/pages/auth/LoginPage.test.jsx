import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { configureStore } from "@reduxjs/toolkit";
import authSlice, {
  setErrorMessage,
} from "../../../../src/app/features/auth/authSlice";
import { initialState } from "../../../fixtures/auth/authFixtures";
import LoginPage from "../../../../src/UI/pages/auth/LoginPage";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { user } from "../../../fixtures/auth/userFixtures";

const mockStartLogin = jest.fn();

jest.mock("../../../../src/app/features/auth/authThunks", () => ({
  startLogin:
    ({ email, pass }) =>
    () =>
      mockStartLogin({ email, pass }),
}));

const mockStore = configureStore({
  reducer: { auth: authSlice },
  preloadedState: {
    auth: initialState,
  },
});

describe("LoginPage", () => {
    beforeEach(() => jest.clearAllMocks());

    test("renders login form", async () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Provider store={mockStore}>
                    <LoginPage />
                </Provider>
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test("submits login form with valid credentials", () => {
        const { email, pass } = user;
        render(
            <MemoryRouter>
                <Provider store={mockStore}>
                    <LoginPage />
                </Provider>
            </MemoryRouter>
        );

        fireEvent.change(screen.getByTestId("email"), {
            target: { value: email },
        });
        fireEvent.change(screen.getByTestId("password"), {
            target: { value: pass },
        });

        fireEvent.click(screen.getByTestId("login-button"));

        expect(mockStartLogin).toHaveBeenCalledWith({ email, pass });
    });

    test("displays error message for invalid credentials", async () => {
        const expectErrorMessage = "Credenciales no v√°lidas";
        mockStartLogin.mockReturnValue(
            mockStore.dispatch(setErrorMessage(expectErrorMessage))
        );

        render(
            <MemoryRouter>
                <Provider store={mockStore}>
                    <LoginPage />
                </Provider>
            </MemoryRouter>
        );

        expect(screen.getByTestId("errorMessage")).toHaveTextContent(
            expectErrorMessage
        );
    });
});