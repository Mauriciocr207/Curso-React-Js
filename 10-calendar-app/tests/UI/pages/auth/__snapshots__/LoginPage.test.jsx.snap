// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders login form 1`] = `
<DocumentFragment>
  <section
    class="bg-gray-100 dark:bg-gray-900"
  >
    <div
      class="flex flex-col items-center justify-center px-6 py-8 mx-auto h-screen lg:py-0-w-[30rem]"
    >
      <h1
        class="text-3xl font-bold leading-tight tracking-tight text-gray-700 md:text dark:text-white mb-5"
      >
        Calendar App
      </h1>
      <div
        class="w-full bg-white rounded-xl shadow-lg dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
      >
        <div
          class="p-6 space-y-4 md:space-y-6 sm:p-8"
        >
          <h2
            class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
          >
            Sign in to your account
          </h2>
          <form
            class="space-y-4 md:space-y-6"
          >
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full shadow rounded-medium"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="relative w-full inline-flex tap-highlight-transparent shadow-sm px-3 border-medium border-default-200 data-[hover=true]:border-default-400 group-data-[focus=true]:border-default-foreground min-h-10 rounded-medium flex-col items-start justify-center gap-0 transition-background !duration-150 transition-colors motion-reduce:transition-none h-14 py-2"
                data-slot="input-wrapper"
                style="cursor: text;"
              >
                <label
                  class="absolute z-10 pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 cursor-text will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-default-600 group-data-[filled-within=true]:pointer-events-auto group-data-[filled-within=true]:scale-85 text-small group-data-[filled-within=true]:-translate-y-[calc(50%_+_theme(fontSize.small)/2_-_6px_-_theme(borderWidth.medium))] pe-2 max-w-full text-ellipsis overflow-hidden"
                  data-slot="label"
                  for="react-aria-:r0:"
                  id="react-aria-:r1:"
                >
                  Your Email
                </label>
                <div
                  class="inline-flex w-full items-center h-full box-border group-data-[has-label=true]:items-end"
                  data-slot="inner-wrapper"
                >
                  <input
                    aria-label="Your Email"
                    aria-labelledby="react-aria-:r0: react-aria-:r1:"
                    class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
                    data-slot="input"
                    data-testid="email"
                    id="react-aria-:r0:"
                    name="email"
                    placeholder="Enter your email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full shadow rounded-medium"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="relative w-full inline-flex tap-highlight-transparent shadow-sm px-3 border-medium border-default-200 data-[hover=true]:border-default-400 group-data-[focus=true]:border-default-foreground min-h-10 rounded-medium flex-col items-start justify-center gap-0 transition-background !duration-150 transition-colors motion-reduce:transition-none h-14 py-2"
                data-slot="input-wrapper"
                style="cursor: text;"
              >
                <label
                  class="absolute z-10 pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 cursor-text will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-default-600 group-data-[filled-within=true]:pointer-events-auto group-data-[filled-within=true]:scale-85 text-small group-data-[filled-within=true]:-translate-y-[calc(50%_+_theme(fontSize.small)/2_-_6px_-_theme(borderWidth.medium))] pe-2 max-w-full text-ellipsis overflow-hidden"
                  data-slot="label"
                  for="react-aria-:r5:"
                  id="react-aria-:r6:"
                >
                  Password
                </label>
                <div
                  class="inline-flex w-full items-center h-full box-border group-data-[has-label=true]:items-end"
                  data-slot="inner-wrapper"
                >
                  <input
                    aria-label="Password"
                    aria-labelledby="react-aria-:r5: react-aria-:r6:"
                    class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
                    data-has-end-content="true"
                    data-slot="input"
                    data-testid="password"
                    id="react-aria-:r5:"
                    name="pass"
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                  <button
                    aria-label="toggle password visibility"
                    class="focus:outline-none"
                    type="button"
                  >
                    <svg
                      class="text-2xl text-default-400 pointer-events-none"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 h-10 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover w-full text-white bg-primary-600 hover:bg-primary-700 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
              data-testid="login-button"
              type="submit"
            >
              <div
                aria-label="Loading"
                class="relative inline-flex flex-col gap-2 items-center justify-center"
              >
                <div
                  class="relative flex w-5 h-5"
                >
                  <i
                    class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent border-2 border-b-white"
                  />
                  <i
                    class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent border-2 border-b-white"
                  />
                </div>
              </div>
            </button>
            <p
              class="text-sm font-light text-gray-500 dark:text-gray-400"
            >
              Donâ€™t have an account yet? 
              <a
                class="font-medium text-primary-600 hover:underline dark:text-primary-500"
                href="/auth/register"
              >
                Sign up
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
